api/v1alpha1/common.go

```diff
@@ -297,32 +297,42 @@ func Match(fc map[string]FilterCond, filter Filter, funcs ...FilterFunc) ([]int,
 		funcs = defaultFilterFuncs
 	}
 
-	// If the chart package name is not in the fc or filterCond.Enabled=false, we will keep all versions of the chart package.
+	// When the chart package name is not in the fc, we will keep all versions of the chart package.
 	filterCond, ok := fc[filter.Name]
-	if !ok || !filterCond.Enabled {
-		for i := range filter.Versions {
-			versions = append(versions, i)
-		}
+	if !ok {
 		return versions, true
 	}
-
-	// If filterCond.Enabled=true and filterCond.VersionedFilterCond=nil, we will not keep all versions of the chart package.
-	if filterCond.Enabled && filterCond.VersionedFilterCond == nil {
-		return versions, false
+	// If filterCond.VersionedFilterCond=nil, it means that
+	// we determine whether to keep the version of the package based on filterCond.KeepDeprecated.
+	if filterCond.VersionedFilterCond == nil {
+		if filterCond.KeepDeprecated {
+			for i, v := range filter.Versions {
+				if v.Deprecated {
+					versions = append(versions, i)
+				}
+			}
+		}
+		return versions, true
 	}
-
-	// If any filter function is satisfied, we will not keep the version.
-	// Then if filterCond.deprecated=false, we will also not keep the deprecated version.
+	// If operation=keep, a version can be kept as long as it satisfies a certain filter function
+	// If operation=ignore, a version is kept only if all filter functions are not satisfied.
+	// Then, based on filterCond.KeepDeprecated, determine whether to keep the version or not.
 	for i, v := range filter.Versions {
 		keep = true
 		for _, f := range funcs {
 			if f(filterCond, v.Version) {
-				keep = false
-				break
+				if filterCond.Operation == FilterOpKeep {
+					keep = true
+				}
+				if filterCond.Operation == FilterOpIgnore {
+					keep = false
+				}
```

Comment:
For filtering conditions, `it is sufficient to satisfy one`, so you can just jump out of the loop if you satisfy the condition here.